clinica/iotools/utils/pipeline_handling.py

```diff
@@ -47,8 +47,67 @@ def pet_volume_pipeline(
     )
 
 
-def t1_freesurfer_pipeline(caps_dir, df, freesurfer_atlas_selection=None, **kwargs):
-    """Merge the data of the PET-Volume pipeline to the merged file containing the BIDS information.
+def dwi_dti_pipeline(
+    caps_dir: str, df: pd.DataFrame, dti_atlas_selection=None, **kwargs
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+
+    from pathlib import Path
+
+    # Ensures that df is correctly indexed
+    if "participant_id" in df.columns.values:
+        df.set_index(
+            ["participant_id", "session_id"], inplace=True, verify_integrity=True
+        )
```

Comment:
I wonder why is `set_index` conditional here, given we are iterating on index values a couple of lines below? We should probably run it unconditionnally and `try / catch` if the input dataframe is not in the expected shape.

Reply:
Correct. That was a mere copy paste from previous code. Granted it could be optimized.